Sources/Utility/ArgumentParser.swift

```diff
@@ -200,6 +203,26 @@ public struct PathArgument: ArgumentKind {
     public static var completion: ShellCompletion = .filename
 }
 
+public struct ExecutableName: ArgumentKind {
+    public let name: String
+
+    public init(argument: String) throws {
+        name = argument
+    }
+
+    public static var completion: ShellCompletion = .function("_swift_executable")
```

Comment:
It needs to work with instances of `ArgumentProtocol`, so an instance variable will not work. We could add the completion to instances of `ArgumentProtocol`, but than the coupling between the ArgumentKind and it’s completions isn’t explicit.

I designed the current implementation such that the ArgumentKind is responsible for providing it’s completions. That’s also why I think that adding two `ArgumentKind`s is no such a big deal, but rather a feature — making the argument’s kind even more explicit.